doc/development/testing.rst

```diff
@@ -0,0 +1,206 @@
+This package already includes some c++ and python code to write tests for.
+
+Writing c++ unit tests
+^^^^^^^^^^^^^^^^^^^^^^
+Tests are always located in the ``test/`` directory of a package. To create a
+test we create a new cpp file in the ``test`` directory called ``AddTest.cpp``
+with the following contents.
+
+.. code::
+
+    #include <gtest/gtest.h>
+
+    #include "ros_test_tutorial/Add.h"
+
+    TEST(AddTest, addTwoAndOne)
+    {
+        Add add;
+        ASSERT_EQ(add.add(2, 1), 3);
+    }
+
+    int main(int argc, char* argv[])
+    {
+        testing::InitGoogleTest(&argc, argv);
+        return RUN_ALL_TESTS();
+    }
+
+Feel free to experiment with writing tests. The actual test is written using
+the ``TEST`` macro.  The first argument is the test suite, the same as the
+filename in this case.  The second argument is name of the actual test. So this
+would be another example of a test.
+
+.. code::
+
+    TEST(AddTest, addNegative)
+    {
+        Add add;
+        ASSERT_EQ(add.add(0, -1), -1);
+    }
+
+In order to build and run the unit tests we must add ``rosunit`` as test
+dependency to our package. So normally we would add the following to the
+``package.xml``. However, this has already been done in highlevel
+``package.xml``.
+
+.. code::
+
+  <test_depend>rosunit</test_depend>
+
+Next we must tell cmake which tests it has to build. So add the following to the ``CMakeLists.txt``.
+
+.. code::
+
+    if(CATKIN_ENABLE_TESTING)
+        catkin_add_gtest(add_test test/AddTest.cpp)
+        target_link_libraries(add_test ros_test_tutorial ${catkin_LIBRARIES})
+    endif()
+
+Here we tell ``cmake`` to build the test and use our library. Now when
+``colcon test`` is run from the workspace root you should see tests passing.
+
+Writing c++ node tests
+^^^^^^^^^^^^^^^^^^^^^^
+
+
+Writing python unit tests
+^^^^^^^^^^^^^^^^^^^^^^^^^
+Tests for python are also always located in the ``test/`` directory. We will
+first start by writing our test. So create the ``MultiplyTest.py`` file to your
+``test/`` directory and fill it with the following contents.
+
+.. code::
+
+    #!/usr/bin/env python
+
+    import unittest
+    import rosunit
+
+    from ros_test_tutorial import multiply
+
+    class MultiplyTest(unittest.TestCase):
+
+        def test_multiply_one_and_one():
+            self.assertEqual(multiply(1, 1), 1)
+
+    if __name__ == '__main__':
+        rosunit.unitrun(PKG, 'test_multiply', MultiplyTest)
+
+The actual tests are written as functions inside the ``MultiplyTest`` class.
+See if you can add more tests (and make them fail). See the `python unittest
+documentation <https://docs.python.org/3/library/unittest.html>`_ for more
+information on writing tests.
+
+Next we must tell ``cmake`` to build and how to run the tests. We do this by
+adding ``catkin_add_nosetests()`` inside our ``CATKIN_ENABLE_TESTING``.
+
+.. code::
+
+    if(CATKIN_ENABLE_TESTING)
+        ...
+        catkin_add_nosetests(test/MultiplyTest.py)
```

Comment:
So this command is for python what 
catkin_add_gtest(add_test test/AddTest.cpp)
target_link_libraries(add_test ros_test_tutorial ${catkin_LIBRARIES})
is for c++?